src/working/merge_sort.py

```diff
@@ -0,0 +1,76 @@
+"""Merge sort data structure."""
+
+
+def mergesort(data):
+    """Sorting algorithm for merg sort."""
+    try:
+        if len(data) > 1:
+            half = len(data) // 2
+            left = data[:half]
+            right = data[half:]
+
+            mergesort(left)
+            mergesort(right)
+
+            di = 0
+            li = 0
+            ri = 0
+
+            while len(left) > li and len(right) > ri:
+                if not isinstance(left[li], int):
+                    raise TypeError()
+                if not isinstance(right[ri], int):
+                    raise TypeError()
```

Comment:
It would be more helpful to your users if they had a message with the Error raised.